File: matiasmasso/Mat.NET/DT2/DTOContractCodi.vb
Public Class DTOContractCodi
    Inherits DTOBaseGuid

    Public Enum wellknowns
        notSet
        reps
    End Enum

    Property nom As String
    Property amortitzable As Boolean

    Public Sub New()
        MyBase.New()
    End Sub

    Public Sub New(oGuid As Guid)
        MyBase.New(oGuid)
    End Sub

    Shared Function wellknown(id As DTOContractCodi.wellknowns) As DTOContractCodi
        Dim retval As DTOContractCodi = Nothing
        Dim sGuid As String = ""
        Select Case id
            Case DTOContractCodi.wellknowns.Reps
                sGuid = "53FA6E40-89BA-4485-8715-F6EB2FCB990E"
        End Select

        If sGuid > "" Then
            Dim oGuid As New Guid(sGuid)
            retval = New DTOContractCodi(oGuid)
        End If
        Return retval
    End Function
End Class

File: matiasmasso/Mat.NET/DTO/DTOContractCodi.cs
using System;

namespace DTO
{
    public class DTOContractCodi : DTOBaseGuid
    {
        public enum Wellknowns
        {
            NotSet,
            Reps
        }

        public string Nom { get; set; }
        public bool Amortitzable { get; set; }
        public DTOContract.Collection Contracts { get; set; }

        public DTOContractCodi() : base()
        {
        }

        public DTOContractCodi(Guid oGuid) : base(oGuid)
        {
        }

        public static DTOContractCodi Wellknown(DTOContractCodi.Wellknowns id)
        {
            DTOContractCodi retval = null;
            string sGuid = "";
            switch (id)
            {
                case DTOContractCodi.Wellknowns.Reps:
                    {
                        sGuid = "53FA6E40-89BA-4485-8715-F6EB2FCB990E";
                        break;
                    }
            }

            if (sGuid.isNotEmpty())
            {
                Guid oGuid = new Guid(sGuid);
                retval = new DTOContractCodi(oGuid);
            }
            return retval;
        }
    }
}
